### ISSUE TITLE
Running 'gradlew wrapper --gradle-version' for a not existing version corrupts the wrapper.properties file

### ISSUE THREAD

**Issue by davidmoshal** [gradle/gradle#16773]
### Expected Behavior

Upgrading to version 7.0 using wrapper task should succeed.

According to the instructions in this link: https://docs.gradle.org/current/userguide/upgrading_version_6.html

```
$ ./gradlew wrapper --gradle-version=7.0 --distribution-type=bin
```

### Current Behavior

The result is:

```
Exception in thread "main" java.io.FileNotFoundException: https://downloads.gradle-dn.com/distributions/gradle-7.0.0-bin.zip
```

### Context

It appears that gradle inserts the extra digit (ie: 7.0.0) and then finds nothing at that link.

ie: instead of looking for the specified 7.0 release, the gradle daemon looks for: 7.0.0 and finds nothing.

### Steps to Reproduce 

as above

### Your Environment

Build scan URL:

not relevant, there is nothing at this link:
- https://downloads.gradle-dn.com/distributions/gradle-7.0.0-bin.zip

Please don't publish new versions without first doing basic QA !

**Comment by melix** [comment]
What version of Gradle are you upgrading from? What is the original contents of your `gradle/wrapper/gradle-wrapper.properties` file? I have created a sample project with 6.8.3, the link is:

```
distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-bin.zip
```

then upgraded to 7.0:

> ./gradlew wrapper --gradle-version=7.0 --distribution-type=bin

and the link is correct:

```
distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
```

Note that yours has `download.gradle-dn.com` which is not the official URL.

> Please don't publish new versions without first doing basic QA !

This tone is not accepted here. Please follow our [code of conduct](https://github.com/gradle/.github/blob/b08eae7ad6da2f6a0e1d2a8faed324c4b2b229dc/CODE_OF_CONDUCT.md). You're very well accepted to improve our test coverage, maybe you'll change your mind regarding what "QA" means.

**Comment by jjohannes** [comment]
@davidmoshal did you maybe accidentally run `./gradlew wrapper --gradle-version=7.0.0` first?

I assume this is what happened. And this is a real issue with how the wrapper works today. If you define a version that does not exists, the `wrapper.properties` files is effectively "corrupt" and you need to edit it manually. So in your case you have to manually change:

```
distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.0-bin.zip
```

to

```
distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
```

I didn't find an existing issue for this, so I am adjusting the title of this one to reflect the problem.

**Comment by dinomite** [comment]
Voting to close this—I had accidentally run `./gradlew wrapper --gradle-version=7.0.0` as suggested by @jjohannes. Undoing the changes to the `wrapper.properties` caused by this and running `./gradlew wrapper --gradle-version=7.0` worked.

**Comment by ljacomet** [comment]
This will be fixed in Gradle 8.2
* See #24010
